File: PAG. CONSULTA DE ESTOQUE/fetch_products.py
import sqlite3
import json

# Conectar ao banco de dados
conn = sqlite3.connect('estoque.db')

# Criar um cursor
cursor = conn.cursor()

# Buscar dados da tabela de produtos
cursor.execute('SELECT nome, categoria, quantidade, preco FROM produtos')
produtos = cursor.fetchall()

# Fechar a conexão
conn.close()

# Converter os dados para JSON
produtos_json = json.dumps([{
    'nome': produto[0],
    'categoria': produto[1],
    'quantidade': produto[2],
    'preco': produto[3]
} for produto in produtos])

# Salvar o JSON em um arquivo
with open('produtos.json', 'w') as f:
    f.write(produtos_json)

File: PAG. CONSULTA DE ESTOQUE/fetch_products.php
<?php
include 'db_connect.php';

$sql = "SELECT nome, categoria, quantidade, preco FROM produtos";
$result = $conn->query($sql);

$produtos = array();
if ($result->num_rows > 0) {
    while($row = $result->fetch_assoc()) {
        $produtos[] = $row;
    }
}

$conn->close();

header('Content-Type: application/json');
echo json_encode($produtos);
?>